### ISSUE TITLE
UIWebView Warning: Update pods of firebase plugin(ITMS-90809 warning)

### ISSUE THREAD

**Issue by Sagarkhenat** [arnesson/cordova-plugin-firebase#1129]
**Describe the bug**
After adding the cordova-plugin-firebase plugin to our ionic hybrid app, I received the warning of ITMS-90809 Deprecated API usage from the apple team after uploading the build on testflight.

When I did the grep -r "UIWebView" . inside the platforms folder it gave me the result of Crashlytics.framework and FirebaseAuth.framework match. I wanted to know how can we check the framework version within the plugin and if I wanted to updated the pods for the framework,how should I be able to do it. Any kind of information regarding this would be helpful.

**Comment by rajashekaranugu** [comment]
any updates on this???

**Comment by IgorVNC** [comment]
Seems a possible solution is to manuallt upgrade thwe following frameworks:

- https://github.com/arnesson/cordova-plugin-firebase/tree/master/src/ios/Crashlytics  Crashlytics.framework here
- https://github.com/arnesson/cordova-plugin-firebase/tree/master/src/ios/Firebase/Auth  FirebaseAuth.framework

which are both 2 years outdated 

As there is not much activity anymore here, it's all up to us to do it

**Comment by Sagarkhenat** [comment]
Hi @IgorVNC , have you tried upgrading the frameworks of this plugin manually? Because we have used this plugin for our app, and it is giving us the UIWebView warning mail.

**Comment by teknepal** [comment]
We switched over to https://github.com/dpa99c/cordova-plugin-firebasex to fix the UIWebView warnings.

**Comment by marutifh** [comment]
> Seems a possible solution is to manually upgrade the iOS following frameworks:
> 
> * https://github.com/arnesson/cordova-plugin-firebase/tree/master/src/ios/Crashlytics  Crashlytics.framework here
> * https://github.com/arnesson/cordova-plugin-firebase/tree/master/src/ios/Firebase/Auth  FirebaseAuth.framework
> 
> which are both 2 years outdated
> 
> As there is not much activity anymore here, it's all up to us to do it

@IgorVNC  Could you just update the PODs?? I am unable to use the plugin mentioned by @teknepal  as I am running on Cordova 7.X.X version, Any help will be appreciated.

**Comment by assertdesignuk** [comment]
We use a forked version the plugin so it works with PGB and xcode, so I am going to update these PODS on our fork and see if it works.

I will report back here.

**Comment by marutifh** [comment]
@assertdesignuk Great! will be waiting for your response.

**Comment by marutifh** [comment]
@assertdesignuk Any luck with the forked version?

**Comment by syedmaqbool** [comment]
> We use a forked version the plugin so it works with PGB and xcode, so I am going to update these PODS on our fork and see if it works.
> 
> I will report back here.

any luck ?

**Comment by marutifh** [comment]
@syedmaqbool  @assertdesignuk 

I created a forked version of cordova-plugin-fcm (This supports push notification but not analytics) 

Here is the link:[https://github.com/marutifh/hrs-cordova-plugin-fcm]( https://github.com/marutifh/hrs-cordova-plugin-fcm)

You may have to change your code to have it working.

**Comment by nieok** [comment]
Has anyone found a solution to this. we are still unable to upload to app store because of the uiwebview references. thanks

**Comment by syedmaqbool** [comment]
> Has anyone found a solution to this?
> We are still unable to upload to app store because of the uiwebview references. thanks

Yes I have found the solution remove firebase plugin and add fcm plugin for push notification

**Comment by nieok** [comment]
thank you for the advice. But i need firebase analytics too. and i need support for remote building like phonegap build or appflow. I prefer to keep using this one with a fix if available.

**Comment by sametacar** [comment]
still no update for uiwebview problem? 
I've some troubles about firebasex like cocoapods root problems etc.
